File: lista3/zadanie2/Pieciokat.cpp
#include "Figura.hpp"
#include "Pieciokat.hpp"
#include <cmath>


double Pieciokat::obliczPole(){
    if (bok <= 0) 
        throw std::invalid_argument(std::to_string(bok)+ std::string("-> Otrzymano niedodatni bok"));
    return (5/4.0) * bok * bok * sqrt(3.0);
}

double Pieciokat::obliczObwod(){
    if (bok <= 0) 
        throw std::invalid_argument(std::to_string(bok) + std::string("-> Otrzymano niedodatni bok"));
    return 5*bok;
}

void Pieciokat::dajNazwe(){
    std::cout << "Jestem pieciokatem" << std::endl;
}

File: lista3/zadanie1/Pieciokat.java
public class Pieciokat extends Figura {

    private int bok;

    Pieciokat(int bok){
        if (bok < 0) throw new IllegalArgumentException(bok + "-> niedodatni bok :c");
        this.bok = bok;
    }

    public int dajBok(){
        return bok;
    }

    public double obliczPole(){
        return ( ( 5.0 / 4.0 ) * bok * bok * Math.sqrt(3.0)); 
    }

    public double obliczObwod(){
        return (5 * bok);
    }

    public void dajNazwe(){
        System.out.println("Jestem pieciokatem");
    }
}
